src/parser/Literal.cpp

```diff
@@ -149,4 +149,19 @@ void Literal::setSubstr(std::size_t start, std::size_t length) {
 // __________________________________________
 void Literal::removeDatatypeOrLanguageTag() { content_.erase(beginOfSuffix_); }
 
+// __________________________________________
+void Literal::replaceContent(const std::string& newContent) {
```

Comment:
Make the argument a `std::string_view` then it beomes
1. more flexible
2. more efficient (you call `substr` on it below).